xcvm/lib/core/src/instruction.rs

```diff
@@ -0,0 +1,34 @@
+use crate::Program;
+use alloc::{collections::VecDeque, vec::Vec};
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+
+/// Base XCVM instructions.
+/// This set will remain as small as possible, expressiveness must come on `top` of the base
+/// instructions.
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+#[derive(Clone, PartialEq, Eq, Debug, Encode, Decode, TypeInfo, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Instruction<Network, Payload, Account, Assets> {
+	/// Transfer some [`Assets`] from the current program to the [`to`] account.
+	#[serde(rename_all = "snake_case")]
+	Transfer { to: Account, assets: Assets },
+	/// Arbitrary payload representing a raw call inside the current [`Network`].
+	///
+	/// On picasso, this will be a SCALE encoded dispatch call.
+	/// On ethereum, an ethereum ABI encoded call.
+	/// On cosmos, a raw json WasmMsg call.
+	///
+	/// Depending on the network, the payload might be more structured than the base call.
+	/// For most of the network in fact, we need to provide the target address along the payload,
+	/// which can be encoded inside this single payload.
+	#[serde(rename_all = "snake_case")]
+	Call { encoded: Payload },
```

Comment:
Not having `address` field here results in platform-specific implementation in the contract side. For example we expect
```json
{
    "address": "blabla",
    "payload": "{Json Encoded WasmMsg}"
}
```
in our cosmwasm interpreter contract. We might have optional `address` field, which would be deducted when serialized if there is no address specified so that we only pay for what we use.

Reply:
Actually, current implementation of the interpreter accepts only `msg` field of `WasmMsg`. I think it should be the whole `WasmMsg` and it should only accept the `Execute` variant of `WasmMsg` which is:
```rust
pub enum WasmMsg {
    Execute {
        contract_addr: String,
        msg: Binary,
        funds: Vec,
    },
    ...
}
```
This already contains the `contract_addr` so we won't need a seperate address field.

Reply:
We now expect `CosmosMsg` in `Call` instruction so this is not a problem anymore :)